src/app/article/page.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import React from 'react';
+import { useNewsContext } from '@/context/news.context';
+import getHumanDate from '@/utils/getHumanDate';
+import styles from './ArticlePage.module.scss';
+
+type Props = {
+  searchParams: { id: string };
+};
+
+function ArticlePage({ searchParams }: Props) {
+  const { news } = useNewsContext();
+  let article = news.find((el) => el.id === searchParams.id)!;
```

Comment:
You know TS was yelling at you for a reason before you added the `!`, if I try to open http://localhost:3000/article?id=12234 in a browser that doesn’t have anything in the localStorage, what will happen? There would be an error and your app is crashed. I can see 2 possible solutions here. First one is the easiest, if there is no article just redirect it to the home page. Second one is you can add a q parameter to the url so you’ll have something like that `[http://localhost:3000/?q=apple/article?id=12`](http://localhost:3000/?q=apple/article?id=12%60) and if there is no article and news you can make a request from a url and pick an article from it.